lib/jxl/enc_frame.cc

```diff
@@ -213,7 +213,7 @@ uint64_t FrameFlagsFromParams(const CompressParams& cparams) {
   // noise is stored within the compressed image and adding noise makes things
   // worse.
   if (ApplyOverride(cparams.noise, dist >= kMinButteraugliForNoise) ||
-      cparams.photon_noise_iso > 0) {
+      cparams.photon_noise_iso > 0 || cparams.manual_noise.size() > 0) {
```

Comment:
Also here (and maybe, for both options, a warning/error if their size is incorrect)

Reply:
Modified the cparams PostInit to check this and return error if it's incorrect.